### PR TITLE
INGM-391 Implement PDO

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Implement the PDO functionalities.

Fixes # INGM-391

### Type of change

- Create the PDONetworkManager class
- Create the ProcessDataThread class
- Add tests

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] USe [type hints](https://docs.python.org/3/library/typing.html) for every function and argument.
- [x] Build documentation locally to verify changes.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingeniamotion tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: ingeniamotion/pdo.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    def start_pdos(
+        self,
+        interface_name: str,
+        refresh_rate: Optional[float] = None,
+    ) -> None:
+        """
+        Start the PDO exchange process.
+
+        Args:
+            interface_name: The interface name where the slaves are connected to.
```
Comment (GullinBustin) [inline ingeniamotion/pdo.py:255]: Working with the interface name is a pain, maybe we must make a decision and only allow one EtherCAT network. Then you always will know the Network.

Comment (GullinBustin) [reply ingeniamotion/pdo.py:255]: But that will have a problem, how will we work when CANopen supports PDOs? it must be defined.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/pdo.py (hunk 1 above)
Comment (MartinALL91) [reply ingeniamotion/pdo.py]: Done

File: ingeniamotion/pdo.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+        """
+        drive = self.mc._get_drive(servo)
+        if not isinstance(drive, EthercatServo):
+            raise ValueError(f"Expected an EthercatServo. Got {type(drive)}")
+        drive.reset_rpdo_mapping()
+        drive.reset_tpdo_mapping()
+
+    def start_pdos(
+        self,
+        network_type: Optional[str] = None,
```
Comment (GullinBustin) [inline ingeniamotion/pdo.py:249]: Don't use string, use enums: https://github.com/ingeniamc/ingeniamotion/blob/master/ingeniamotion/enums/__init__.py#L148

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/pdo.py (hunk 2 above)
Comment (MartinALL91) [reply ingeniamotion/pdo.py:249]: It was my first thought, but I think it would be more complicated for the user.

Comment (MartinALL91) [reply ingeniamotion/pdo.py]: Done

**Review by WoodElv — APPROVED** [review]

File: ingeniamotion/motion_controller.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 from typing import Dict
 
 from ingenialink.network import Network
+from ingenialink.ethercat.network import EthercatNetwork
```
Comment (GullinBustin) [inline ingeniamotion/motion_controller.py:5]: Why did you add this?

File: ingeniamotion/pdo.py [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+        mc: The MotionController.
+
+    """
+
+    class ProcessDataThread(threading.Thread):
+        """Manage the PDO exchange.
+
+        Attributes:
+            net: The EthercatNetwork instance where the PDOs will be active.
+            refresh_rate: Determines how often (seconds) the PDO values will be updated.
```
Comment (GullinBustin) [inline ingeniamotion/pdo.py:34]: notify_send_process_data and notify_receive_process_data are missing. And they are Args, not Attributes

File: ingeniamotion/pdo.py [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+            if refresh_rate is None:
+                refresh_rate = self.DEFAULT_PDO_REFRESH_RATE
+            elif refresh_rate > self.MINIMUM_PDO_REFRESH_RATE:
+                raise ValueError(
+                    f"The minimum PDO refresh rate is {self.MINIMUM_PDO_REFRESH_RATE} seconds."
+                )
+            self._refresh_rate = refresh_rate
+            self._pd_thread_stop_event = threading.Event()
+            for servo in self._net.servos:
+                servo.slave.set_watchdog(self.ETHERCAT_PDO_WATCHDOG, self._refresh_rate * 1500)
```
Comment (GullinBustin) [inline ingeniamotion/pdo.py:60]: Use constant instead a literal 1500

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/motion_controller.py (hunk 3 above)
Comment (MartinALL91) [reply ingeniamotion/motion_controller.py]: By mistake. Done

**Review by pzinemanas — APPROVED** [review]

**Review by GullinBustin — APPROVED** [review]
